bundles/CoreBundle/Templating/LegacyTemplateGuesser.php

```diff
@@ -47,10 +47,8 @@ public function __construct(KernelInterface $kernel, DelegatingEngine $templateE
      */
     public function guessTemplateName($controller, Request $request, $engine = 'twig')
     {
-        if ($request->attributes->has('_template')) {
-            if ($template = $request->attributes->get('_template') instanceof TemplatePhp) {
-                $engine = 'php';
-            }
+        if ($request->attributes->has('_template') && $request->attributes->get('_template') instanceof TemplatePhp) {
```

Comment:
Another possibility would be to use the real `engine` value of the annotation instead:
```suggestion
        if ($request->attributes->has('_template')) {
            $template = $request->attributes->get('_template');

            if ($template instanceof \Sensio\Bundle\FrameworkExtraBundle\Configuration\Template) {
                $engine = $template->getEngine();
            }
        }
```

Reply:
$engine property is removed from \Sensio\Bundle\FrameworkExtraBundle\Configuration\Template in 5.x version as now only Twig is supported so this will not work.

Reply:
Oh okay, I didn't think of that.